ehcache-impl/src/main/java/org/ehcache/impl/internal/loaderwriter/writebehind/BatchingLocalHeapWriteBehindQueue.java

```diff
@@ -188,13 +193,19 @@ public boolean add(SingleOperation<K, V> operation) {
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public void run() {
       try {
         List<BatchOperation<K, V>> batches = createMonomorphicBatches(operations());
         // execute the batch operations
         for (BatchOperation<K, V> batch : batches) {
           try {
             batch.performOperation(cacheLoaderWriter);
+          }  catch (BulkCacheWritingException e) {
+            Iterator<K> it = (Iterator<K>) e.getFailures().keySet().iterator();
```

Comment:
Always write the simplest code that you can... no need to challenge the runtime more than necessary:
```java
for (Object key : e.getFailures().keySet()) {
  keyCleanUpMethod.accept((K) key);
}
```

Reply:
You also need to think about this generic cast... will there be a concrete cast done somwhere down the line? Where will that ClassCastException go? Do we need to defend against? Good test coverage might be able to protect you here.